### PR TITLE
Use `ensure_source_root` in `darwinia-support`

### PR DESCRIPTION
**Pull request by AsceticBear**

### CODE REVIEW

File: frame/support/src/lib.rs [hunk 1]
```diff
@@ -83,12 +86,18 @@ pub mod s2s {
...
+	pub fn ensure_source_root<AccountId, Converter>(
 		chain_id: ChainId,
-	) -> AccountId {
+		account: &AccountId,
+	) -> Result<(), DispatchError>
+	where
+		AccountId: Encode + sp_std::cmp::PartialEq,
+		Converter: Convert<H256, AccountId>,
+	{
 		let hex_id = derive_account_id::<AccountId>(chain_id, SourceAccount::Root);
```
Comment (AurevoirXavier) [inline frame/support/src/lib.rs:97]: It seems `SouceAccount` is not in this scope.

File: frame/support/src/lib.rs [hunk 2]
```diff
@@ -83,12 +86,18 @@ pub mod s2s {
...
 	}
 
-	pub fn source_root_converted_id<AccountId: Encode, Converter: Convert<H256, AccountId>>(
+	pub fn ensure_source_root<AccountId, Converter>(
 		chain_id: ChainId,
-	) -> AccountId {
+		account: &AccountId,
+	) -> Result<(), DispatchError>
+	where
+		AccountId: Encode + sp_std::cmp::PartialEq,
```
Comment (AurevoirXavier) [inline frame/support/src/lib.rs:94]:
```suggestion
	AccountId: PartialEq + Encode,
```

Comment (AsceticBear) [reply frame/support/src/lib.rs]: Fixed.

File: frame/support/src/lib.rs [hunk 3]
```diff
@@ -31,7 +31,7 @@ pub mod balance {
 }
 
 pub mod evm {
-	// --- darwinia ---
+	// --- crates.io ---
 	use ethereum::TransactionMessage;
 	use ethereum_primitives::{H160, H256};
```
Comment (AurevoirXavier) [inline frame/support/src/lib.rs:36]: darwinia part

File: frame/support/src/lib.rs [hunk 4]
```diff
@@ -59,12 +59,18 @@ pub mod evm {
 
 // TODO: Should we move this to `s2s-primitives`?
 pub mod s2s {
-	use bp_runtime::{derive_account_id, ChainId, SourceAccount};
+	// --- crates.io ---
 	use codec::Encode;
 	use ethabi::{encode, Token};
 	use ethereum_primitives::{H160, H256};
```
Comment (AurevoirXavier) [inline frame/support/src/lib.rs:65]: darwinia part

**Pull request merged**